src/app/inventory/store/d2-item-factory.service.js

```diff
@@ -195,11 +220,19 @@ export function D2ItemFactory(
 
     const itemType = normalBucket.type || 'Unknown';
 
-    // Bungie API Bug: infusionCategoryName is missing as of CoO: https://github.com/Bungie-net/api/issues/324
-    // TODO: We should be figuring out our categories based on
-    // itemDef.itemCategoryHashes instead. Unfortunately they're
-    // localized and don't map cleanly into our existing categories.
-    const categories = itemDef.quality && itemDef.quality.infusionCategoryName ? [itemDef.quality.infusionCategoryName.replace('v300.', 'category_').toUpperCase()] : [];
+    const categories = [];
```

Comment:
This logic would benefit from being refactored into its own method.

Reply:
Once the second half is gone, it is a relatively short piece of code. Would that still benefit?

Reply:
Yeah, we really should be extracting each bit of non-trivial logic into its own function - this whole function is really large and hard to follow.